blocks/rich-text/index.js

```diff
@@ -692,12 +668,28 @@ export class RichText extends Component {
 		} );
 	}
 
-	setContent( content = '' ) {
-		this.editor.setContent( renderToString( content ) );
+	setContent( content ) {
+		const { format } = this.props;
+		switch ( format ) {
+			case 'string':
+				this.editor.setContent( content || '' );
+				break;
+			default:
+				this.editor.setContent( elementToString( content ) );
+		}
 	}
 
 	getContent() {
-		return nodeListToReact( this.editor.getBody().childNodes || [], createTinyMCEElement );
+		const { format } = this.props;
+
+		switch ( format ) {
+			case 'string':
+				return this.editor.getContent();
```

Comment:
Will this be executed many times? This will be much slower than `this.editor.getContent( { format: 'raw' } )` or one loop over `this.editor.getBody().childNodes`.

Reply:
It will be executed on each change (type), It's probably slower but since it's limited to a single RichText, maybe it's not that important.

`raw` is not great because we don't want to save the TinyMCE temporary nodes and attributes. We could probably write an alternative using `childNodes` and stripping these nodes, but not sure if this is different of what's done internally in TinyMCE

Reply:
Yeah, I think this is fine as long as it's used on inline `RichText` instances. Once we get lists to move over to nested blocks we can even get rid of multiline instances altogether.